metrics.go

```diff
@@ -11,32 +11,42 @@ import (
 // with values relative to that last second.
 type MetricsBag map[string]interface{}
 
-// AddRawCounter adds a counter that always increments.
+// AddRawCounter adds a metric of type counter to the bag and sets its current value.
+// A counter is a cumulative metric that can only increase.
+// To be meaningful, `value` must be positive.
 func (bag MetricsBag) AddRawCounter(name string, value int64) {
 	bag["c:"+name] = value
 }
 
-// AddDeltaCounter adds a count of something that happened in the last second
+// AddDeltaCounter adds a metric of type counter to the bag and increments its value.
+// A counter is a cumulative metric that can only increase.
+// To be meaningful, `delta` must be positive.
 func (bag MetricsBag) AddDeltaCounter(name string, delta int64) {
 	bag["d:"+name] = delta
 }
 
-// AddGauge takes a snapshot of a value.
+// AddGauge adds a metric of type gauge to the bag and updates its value.
+// A gauge represents a single numerical data point that can arbitrarily go up and down.
 func (bag MetricsBag) AddGauge(name string, value float64) {
 	bag["g:"+name] = value
 }
 
-// AddHistogram adds a set of values to track their statistical distribution.
+// AddHistogram adds a metrics of type histogram to the bag and tracks its value.
+// A histogram samples observations and counts them in different 'buckets' in order
+// to track and show the statistical distribution of a set of values.
 func (bag MetricsBag) AddHistogram(name string, values []float64) {
 	bag["h:"+name] = values
 }
 
-// AddTimings adds a set of timings to track their statistical distribution.
+// AddTimings adds a metric of type time to the bag and traks its value.
```

Comment:
under the hood, there's no metric of  `time` or `timing`, this is just an wrapper around an histogram tracking time duration, such as latency
```suggestion
// AddTimings adds a metric of type histogram to the bag and tracks its value.
```